pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -538,39 +647,155 @@ func (r *Builder) mapNetworks(vm *model.Workload, object *cnv.VirtualMachineSpec
-					networkPair = &networkMaps[i]
-					if networkPair.Source.ID == vmNetworkID {
-						found = true
-						break
-					}
+			}
+			var networkPair *api.NetworkPair
+			networkMaps := r.Context.Map.Network.Spec.Map
+			found := false
+			for i := range networkMaps {
+				networkPair = &networkMaps[i]
+				if networkPair.Source.ID == vmNetworkID {
+					found = true
+					break
 				}
-				if !found {
-					err = liberr.New("no network map for vm network", "network", vmNetworkID)
-					return
+			}
+			if !found {
+				err = liberr.New("no network map for vm network",
+					"vm", vm.Name,
+					"network", vmNetworkName,
+					"networkID", vmNetworkID)
+				return
+			}
+
+			// Skip network mappings with destination type 'Ignored'
+			if networkPair.Destination.Type == Ignored {
+				continue
+			}
+
+			// Group NICs by MAC address to identify unique NICs
+			// Multiple entries with the same MAC = same NIC with multiple IPs
+			type NICInfo struct {
+				Entry interface{}
+				MAC   string
+				IPs   []string
+			}
+			nicsByMAC := make(map[string]*NICInfo)
+
+			for _, nicEntry := range nics {
+				if m, ok := nicEntry.(map[string]interface{}); ok {
+					// Skip floating IPs
+					if ipType, ok := m[OSExtIPsType]; ok {
+						if ipType.(string) == "floating" {
+							continue
+						}
+					}
+
+					// Get MAC address
+					macAddr := ""
+					if macAddress, ok := m[OSExtIPsMacAddr]; ok {
+						macAddr = macAddress.(string)
+					}
+
+					if macAddr == "" {
+						continue
+					}
+
+					// Get IP address
+					ipAddr := ""
+					if addr, ok := m["addr"]; ok {
+						if addrStr, ok := addr.(string); ok {
+							ipAddr = addrStr
+						}
+					}
+
+					// Track unique NICs by MAC address and collect all IPs
+					if nicsByMAC[macAddr] == nil {
+						nicsByMAC[macAddr] = &NICInfo{
+							Entry: nicEntry,
+							MAC:   macAddr,
+							IPs:   []string{},
+						}
+					}
+					if ipAddr != "" {
+						nicsByMAC[macAddr].IPs = append(nicsByMAC[macAddr].IPs, ipAddr)
+					}
 				}
+			}
+
+			// Process each unique NIC (by MAC address)
+			for _, nicInfo := range nicsByMAC {
+
+				nic := nicInfo.Entry
+
+				// Check if this destination has already been used
+				var destIdentifier string
+				var destNADName string
+				var destNADNamespace string
+
+				if networkPair.Destination.Type == Pod {
+					destIdentifier = "pod"
+					destNADName = ""
+					destNADNamespace = ""
+				} else if networkPair.Destination.Type == Multus {
+					destNADName = networkPair.Destination.Name
+					destNADNamespace = networkPair.Destination.Namespace
+					destIdentifier = path.Join(destNADNamespace, destNADName)
+
+					// If this Multus NAD has already been used, auto-generate a new one
+					if usedDestinations[destIdentifier] {
+						// Save the original NAD name before modification
+						originalDestNADName := destNADName
+
+						// Get the counter for this original NAD
+						originalKey := path.Join(destNADNamespace, originalDestNADName)
```

Comment:
The destiantion namespace might not contian NAD.
The `default` namespace is special for NADs and makes it globally acessible.